### PR TITLE
SALTO-5692: Okta: Upgrade `Domain`, `SmsTemplate`, `DeviceAssurance`, `UserType` deploy definitions to new infra

### PR DESCRIPTION
**Pull request by Nurdok**
This PR upgrades some simple types to the new infra.
Note: I thought `UserType` had a trivial definitions, but it turns out there's a filter to copy its `_links` attribute on addition. This is pretty simple so I still included it in this PR.

---

_Additional context for reviewer_
None

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/test/deploy_mock_replies.json
Comment (shir-reifenberg) [inline packages/okta-adapter/test/deploy_mock_replies.json:1]: can you please replace our okta base url with something else?

File: packages/okta-adapter/test/adapter.test.ts [hunk 1]
```diff
@@ -702,5 +712,306 @@ describe('adapter', () => {
...
+            },
+          },
+        })
+        userType = new InstanceElement('userType', userTypeType, {
+          id: 'usertype-fakeid1',
+          name: 'superuser',
+          [LINKS_FIELD]: {
+            schema: {
+              rel: 'schema',
+              href: 'https://salto.okta.com/api/v1/meta/schemas/user/oscg64q0mq1aYdKLt697',
```
Comment (shir-reifenberg) [inline packages/okta-adapter/test/adapter.test.ts:808]:
```suggestion
              href: 'https://okta.okta.com/api/v1/meta/schemas/user/oscg64q0mq1aYdKLt697',
```

File: packages/okta-adapter/src/definitions/deploy.ts [hunk 2]
```diff
@@ -30,9 +37,44 @@ const createCustomizations = (): Record<string, InstanceDeployApiDefinitions> =>
...
+      requestsByAction: {
+        customizations: {
+          add: [
+            {
+              request: {
+                endpoint: { path: '/api/v1/meta/types/user', method: 'post' },
+              },
+              copyFromResponse: {
+                additional: {
+                  pick: ['_links'],
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/definitions/deploy.ts:56]: nit: use the const?

File: packages/okta-adapter/src/filters/user_type.ts
Comment (shir-reifenberg) [inline packages/okta-adapter/src/filters/user_type.ts:1]: yay 🎉

**Review by shir-reifenberg — APPROVED** [review]
Looks Great!!
just making sure you also checked manually cause some of those types are not covered by e2e

**Comment by Nurdok** [comment]
> Looks Great!! just making sure you also checked manually cause some of those types are not covered by e2e

Thanks! I manually tested all the types in this PR for add, modify and remove.

**Pull request merged**